File: backend/app/schemas/assessment.py
from typing import Literal

from pydantic import BaseModel, Field


RiskLevel = Literal["low", "medium", "high", "critical"]


class HotelProfile(BaseModel):
    business_name: str = Field(min_length=2, max_length=120)
    municipality: str = Field(min_length=2, max_length=80)
    business_type: Literal[
        "hotel",
        "apartahotel",
        "villa",
        "hostal",
        "alquiler_vacacional",
        "agencia_turistica",
    ]
    rooms_count: int = Field(ge=1, le=1000)
    permanent_employees: int = Field(ge=0, le=500)
    temporary_employees: int = Field(ge=0, le=500)
    has_external_it_provider: bool
    uses_pms: bool
    offers_guest_wifi: bool
    handles_card_payments: bool
    stores_guest_documents: bool


class SecurityControls(BaseModel):
    uses_mfa: bool
    uses_password_manager: bool
    shared_accounts: bool
    pms_individual_users: bool
    employee_offboarding_process: bool
    backup_frequency: Literal["none", "monthly", "weekly", "daily"]
    backups_tested: bool
    has_antivirus: bool
    systems_updated: bool
    guest_wifi_separated: bool
    payment_terminal_isolated: bool
    cctv_or_iot_devices: bool
    iot_network_separated: bool
    supplier_remote_access: bool
    supplier_access_controlled: bool
    has_incident_response_plan: bool
    has_rgpd_breach_protocol: bool
    rgpd_processing_register: bool
    staff_phishing_training: bool


class AssessmentRequest(BaseModel):
    hotel_profile: HotelProfile
    security_controls: SecurityControls


class AreaScore(BaseModel):
    area: str
    score: int = Field(ge=0, le=100)


class RiskFinding(BaseModel):
    title: str
    description: str
    severity: RiskLevel
    recommendation: str


class AssessmentResponse(BaseModel):
    business_name: str
    overall_score: int = Field(ge=0, le=100)
    risk_level: RiskLevel
    area_scores: list[AreaScore]
    risks: list[RiskFinding]
    next_steps: list[str]

File: frontend/src/api/assessment.ts
export type BusinessType =
  | 'hotel'
  | 'apartahotel'
  | 'villa'
  | 'hostal'
  | 'alquiler_vacacional'
  | 'agencia_turistica'

export type BackupFrequency = 'none' | 'monthly' | 'weekly' | 'daily'

export type RiskLevel = 'low' | 'medium' | 'high' | 'critical'

export type AssessmentRequest = {
  hotel_profile: {
    business_name: string
    municipality: string
    business_type: BusinessType
    rooms_count: number
    permanent_employees: number
    temporary_employees: number
    has_external_it_provider: boolean
  }
  security_controls: {
    uses_mfa: boolean
    uses_password_manager: boolean
    shared_accounts: boolean
    backup_frequency: BackupFrequency
    backups_tested: boolean
    has_antivirus: boolean
    systems_updated: boolean
    guest_wifi_separated: boolean
    has_incident_response_plan: boolean
    has_rgpd_breach_protocol: boolean
    staff_phishing_training: boolean
  }
}

export type AreaScore = {
  area: string
  score: number
}

export type RiskFinding = {
  title: string
  description: string
  severity: RiskLevel
  recommendation: string
}

export type AssessmentResponse = {
  business_name: string
  overall_score: number
  risk_level: RiskLevel
  area_scores: AreaScore[]
  risks: RiskFinding[]
  next_steps: string[]
}

const API_BASE_URL = import.meta.env.VITE_API_URL ?? 'http://localhost:8000'

export async function analyzeHotelAssessment(
  payload: AssessmentRequest,
): Promise<AssessmentResponse> {
  const response = await fetch(`${API_BASE_URL}/api/v1/assessments/analyze`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(payload),
  })

  if (!response.ok) {
    throw new Error(`Assessment failed with status ${response.status}`)
  }

  return response.json() as Promise<AssessmentResponse>
}
